starknet_py/net/schemas/gateway.py

```diff
@@ -275,61 +278,64 @@ def make_dataclass(self, data, **kwargs):
-    key = Felt(data_key="key", required=True)
-    value = Felt(data_key="value", required=True)
+    class_hash = NonPrefixedHex(data_key="class_hash", required=True)
 
     @post_load
     def make_dataclass(self, data, **kwargs):
-        return StorageDiff(**data)
+        return DeployedContract(**data)
 
 
-class DeployedContractSchema(Schema):
-    address = Felt(data_key="address", required=True)
-    class_hash = NonPrefixedHex(data_key="class_hash", required=True)
+class StateDiffSchema(Schema):
+    deployed_contracts = fields.List(
+        fields.Nested(DeployedContractSchema()),
+        data_key="deployed_contracts",
+        required=True,
+    )
+    declared_contract_hashes = fields.List(
+        Felt(),
+        data_key="declared_contracts",
+        required=True,
+    )
+    storage_diffs = fields.Dict(
+        keys=fields.String(),
+        values=fields.Raw(),
+        data_key="storage_diffs",
+        required=True,
+    )
+    nonces = fields.Dict(keys=Felt(), values=Felt(), data_key="nonces", required=True)
 
     @post_load
-    def make_dataclass(self, data, **kwargs):
-        return DeployedContract(**data)
+    def make_dataclass(self, data, **kwargs) -> StateDiff:
+        return StateDiff(**data)
 
 
 class BlockStateUpdateSchema(Schema):
     block_hash = Felt(data_key="block_hash", required=True)
     new_root = NonPrefixedHex(data_key="new_root", required=True)
     old_root = NonPrefixedHex(data_key="old_root", required=True)
-    state_diff = fields.Dict(
-        keys=fields.String(), values=fields.Raw(), data_key="state_diff", required=True
-    )
+    state_diff = fields.Nested(StateDiffSchema(), data_key="state_diff", required=True)
 
     @post_load
     def make_dataclass(self, data, **kwargs):
-        deployed_contracts = data["state_diff"]["deployed_contracts"]
-        deployed_contracts = [
-            cast(DeployedContract, DeployedContractSchema().load(contract))
-            for contract in deployed_contracts
-        ]
-
-        storage_diffs = []
-        for address, diffs in data["state_diff"]["storage_diffs"].items():
-            for diff in diffs:
-                storage_diff = StorageDiffSchema().load(
-                    {"address": address, "key": diff["key"], "value": diff["value"]}
-                )
-                storage_diffs.append(storage_diff)
-
-        declared_contracts = data["state_diff"]["declared_contracts"]
-        declared_contracts = [
-            int(declared_contract, 16) for declared_contract in declared_contracts
-        ]
-
-        del data["state_diff"]
-
+        storage_diffs: Dict = data["state_diff"].storage_diffs
+        fixed_storage_diffs: List[StorageDiff] = []
+        for address in storage_diffs.keys():
+            entries = []
+            for entry in storage_diffs[address]:
+                entries.append(StorageEntrySchema().load(entry))
+            fixed_storage_diffs.append(StorageDiff(address, entries))
```

Comment:
I know it's been like this before this PR, but I'd consider moving this to a separate method so it's a bit more readable.

Reply:
Sure, makes a lot of sense!